portfolio/src/main/webapp/script.js

```diff
@@ -47,17 +47,100 @@ function newCommentBox(comment){
   return commentBox;
 }
 
+function clearChildren(dom) {
+  while(dom.lastElementChild) {
+    dom.removeChild(dom.lastElementChild);
+  }
+}
+
 /**
- * Adds a greeting to the page using Promises
+ * Loads comments to the page
  */
 function loadComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  const commentContainer = document.getElementById('comment-container');
+  const limit = document.getElementById("limit").value;
+
+  // Clear previous children
+  clearChildren(commentContainer);
+
+  // Repopulate comment section
+  fetch(`/data?limit=${limit}`).then(response => response.json()).then((comments) => {
     for(const comment of comments) {
-      document.getElementById('comment-container').appendChild(newCommentBox(comment));
+      commentContainer.appendChild(newCommentBox(comment));
     }
   });
 }
 
+function verifyDelete() {
```

Comment:
Readability comment: it's not clear from the name of this function how I'm expected to use it. I need to read the function in order to figure out that it returns boolean, and I need to read it closely to figure out that it returns true when I should delete.

For comparison, here is another approach to implementing a verification function

```
function verifyDelete() {
 var inputWord = prompt('What\'s the magic word?');
  if (inputWord != 'please') {
    alert('Comments have been deleted.')
    throw new Error(`User entered ${inputWord}`);
  }
}
```

Then usage would be totally different:

```
function deleteComments() {
  verifyDelete();  // if this throws an exception, we don't execute the rest of the function

  // implementation
}
```

So instead of calling the function "verifyDelete", maybe we can name it something like "shouldDelete". Then it's clear that the function returns a boolean, and if it returns true then we "should delete" it.